server/src/main/java/keywhiz/auth/ldap/LdapAuthenticator.java

```diff
@@ -53,6 +55,10 @@ public LdapAuthenticator(LdapConnectionFactory connectionFactory, LdapLookupConf
 
   @Override
   public Optional<User> authenticate(BasicCredentials credentials) {
+    return trace("LdapAuthenticator.authenticate", () -> doAuthenticate(credentials));
+  }
+
+  private Optional<User> doAuthenticate(BasicCredentials credentials) {
```

Comment:
Rather than wrapping larger methods in a lambda and adding another level of indentation, I find it more readable to just move the logic into a "doX" helper method so that the metrics wrapping is hidden from the actual business logic as much as possible.